dependencies.yaml

```diff
@@ -165,32 +195,82 @@ channels:
 
 dependencies:
 
-  build_cpp: # should be split into `build_python` if/when converting to use scikit-build
+  python:
+    common:
+      - output_types: [conda]
+        packages:
+          - python=3.10
+
+  cudatoolkit:
+    specific:
+      - output_types: [conda]
+        matrices:
+          - matrix:
+              cuda: "12.1"
+            packages:
+              - cuda-cudart=12.1
+              - cuda-nvrtc=12.1
+              - cuda-nvtx=12.1
+              - cuda-version=12.1
+
+  cudatoolkit-dev:
+    specific:
+      - output_types: [conda]
+        matrices:
+          - matrix:
+              cuda: "12.1"
+            packages:
+              - cuda-cudart-dev=12.1
+              - cuda-nvml-dev=12.1
+              - cuda-nvrtc-dev=12.1
+              - cuda-nvtx-dev=12.1
+              - cuda-tools=12.1
+              - cuda-version=12.1
+
+
+
+  # Build dependencies for Morpheus on the target arch. Mirrors the `build` section in
+  # ci/conda/recipes/morpheus/meta.yaml
+  build_cpp_build:
     common:
       - output_types: [conda]
         packages:
-          - boost-cpp=1.84
+          # Compilers
+          - cuda-nvcc=12.1
+          - cxx-compiler
+          - gcc_linux-64=11.2
+          - gxx_linux-64=11.2
+
+          # Non-Compiler Dependencies
+          - automake=1.16.5 # Needed for DOCA build
           - ccache
           - cmake=3.27
-          - cuda-nvcc
+          - cuda-cudart-dev=12.1
+          - cuda-version=12.1
+          - libtool # Needed for DOCA build
+          - ninja=1.11
+          - pkg-config=0.29 # for mrc cmake
+          - sysroot_linux-64=2.17
+
+  # Build dependencies for Morpheus on the host arch. Mirrors the `host` section in
+  # ci/conda/recipes/morpheus/meta.yaml
+  build_cpp_host:
+    common:
+      - output_types: [conda]
+        packages:
+          # Include: cudatoolkit-dev
           - cudf=24.02
-          - cxx-compiler
           - cython=3.0
-          - gcc_linux-64=11.2
           - glog=0.6
-          - gxx_linux-64=11.2
+          - libcudf=24.02
           - librdkafka>=1.9.2,<1.10.0a0
           - mrc=24.06
-          - ninja=1.11
           - nlohmann_json=3.11
-          - pkg-config=0.29 # for mrc cmake
-          - protobuf=4.24
           - pybind11-stubgen=0.10.5
           - rapidjson=1.1.0
+          - rdma-core>=48 # Needed for DOCA.
           - scikit-build=0.17.6
-          - sysroot_linux-64=2.17
-          - tritonclient=2.34
-          - ucx=1.15
+          - versioneer-518
```

Comment:
I think the switch from `versioneer` to `versioneer-518` is causing build failures